app/javascript/packages/lite-webpack-dev-server/webpack-plugin.js

```diff
@@ -0,0 +1,83 @@
+const http = require('node:http');
+const { join } = require('node:path');
+const { createReadStream } = require('node:fs');
+
+/**
+ * @typedef PluginOptions
+ * @prop {string=} publicPath
+ * @prop {number=} port
+ * @prop {Record<string, string>=} headers
+ */
+
+/**
+ * Webpack plugin name.
+ *
+ * @type {string}
+ */
+const PLUGIN = 'LiteWebpackDevServerPlugin';
+
+class LiteWebpackDevServerPlugin {
+  /**
+   * @type {string}
+   */
+  publicPath;
+
+  /**
+   * @type {number}
+   */
+  port;
+
+  /**
+   * @type {Record<string, string>}
+   */
+  headers;
+
+  /**
+   * @param {PluginOptions} options
+   */
+  constructor(options) {
+    Object.assign(this, {
+      publicPath: '.',
+      port: 3035,
+      headers: {
+        'content-type': 'text/javascript',
+        ...options.headers,
+      },
+      ...options,
+    });
+  }
+
+  /**
+   * @param {import('webpack').Compiler} compiler
+   */
+  apply(compiler) {
+    /** @type {Promise<void>} */
+    let build = Promise.resolve();
```

Comment:
are we allowed to `await` on null or undefined? seems weird to create this and wait on it only to override it later

Or is it possible to reorder this so that `compiler.hooks` which assigns `build` gets set up before `http.createServer`?

Reply:
Hm, actually, yeah, we could probably leave this uninitialized and it would work, since `await undefined` is fine.

Originally I was thinking I'd need to be able to support a chain of rebuilds, so having something like `build.then(() => nextBuild)`, but I didn't think that was strictly necessary. The downside is if someone saves multiple times and refreshes the page, the page would load with the result of the first compilation.

So maybe it actually would be nice to have that behavior, but there's a decent amount of complexity in implementing it 😅